ADR Support most recent action images

lucyb opened the pull request:
Describes the decision to support Python v2 rather than Python v1 and the other most recent action images.

Fixes https://github.com/opensafely-core/codespaces-initiative/issues/67

iaindillingham commented on docs/adr/0005-support-most-recent-action-images.md:
```diff
@@ -0,0 +1,29 @@
+
+We want to provide a Codespaces environment that's intuitive for researchers to understand and that provides a "habitable" environment for them in which to write code. We also want to keep the maintenance burden for developers low.
+
+We would also like to encourage researchers to use the newest action images available. We believe that providing a development environment that supports that aim will help with this.
```
I think we should document why we'd like to encourage researchers to use the most recent action images.

iaindillingham commented on docs/adr/0005-support-most-recent-action-images.md:
```diff
@@ -0,0 +1,29 @@
+
+We did a soft rollout of the Python v2 action image earlier this year. At present only one or two researchers are using it, although we expect that number to increase over time. So it's possible that we're making a decision that means the Codespaces experience won't be as good for some older projects. However, we believe those numbers are very low and that the impact of not supporting Python v1 i ...
+
+In future, researchers using older, unsupported action images could edit their dev container config to point to an older Docker image that still uses the older action images. However, we could potentially make other changes that means this doesn't work as expected. Over time, we may find that the Codespaces environment doesn't meet the needs of developers of older research projects. We intend to  ...
```
I don't think older (i.e. not the most recent) action images are unsupported. If a security vulnerability were found, then they would be updated. It's just that they aren't being updated in a backwards-incompatible way. Could you remove the word "unsupported"?

I think we should document how we'd expect a researcher to edit their dev container config to point to an older Docker image. Do we expect them to edit the following line in `devcontainer.json`?

```json
"image": "ghcr.io/opensafely-core/research-template:v0",
```

If we do, then what should it look like?

lucyb replied:
I meant not supported by us with Codespaces rather than the action images themselves being unsupported. I've written it in a confusing way though and I agree about removing it.

Yes, I can be clearer about what editing the dev container config would look like. Good idea.

iaindillingham reviewed (approved):
I've approved, but would really appreciate your thoughts on my comments before you merge.

lucyb commented:
Thanks for the feedback, Iain. I believe I've addressed your comments, so I'll go ahead and merge the changes. However, if you don't think that's the case, please let me know.
